### ISSUE TITLE
CSL-JSON dates are wrong

### ISSUE THREAD

**Issue by larsgw** [larsgw/citation.js#53]
**Problem**

The CSL-JSON date format used in Citation.js is wrong, and therefore not picked up by citeproc-js.

**Consequence**

All dates are omitted in all formatted Citation.js output.

**Cause**

Miscommunication. When designing this version of Citation.js, a different date format was taken into account, as that was the documented format back then.

**Expected solution**

Solution isn't trivial, but not really complex either. Just a lot of time making sure everything is updated. Problem expected to be fixed in `v0.3.0-11`.

---

Thanks to @austinjp for uncovering this issue in #52 and to @fbennett for helping with the citeproc-js side of things.

**Comment by lexnat187** [comment]
@larsgw do you have a projected completion date for v.0.3.0-11?

**Comment by larsgw** [comment]
@lexnat187 Hopefully later today. Surprisingly, the most basic fix already works, but only at output level, so not for `Cite.parse.*()` or the `Cite#data` property yet. Doing that will take a lot longer, because there's a lot more methods that need to change. Test cases also need to be updated. However, that shouldn't take too long, I hope.

**Comment by lexnat187** [comment]
@larsgw that's great news. Let me know when your done and I can test it with my implementation. Let me know if I can help.

**Comment by larsgw** [comment]
Delay and dismay: trouble because of browser distributions. Very annoying. I hope to be able to release it anyway, somewhere before 01:00 UTC.

**Comment by larsgw** [comment]
[`v0.3.0-11` is live](https://www.npmjs.com/package/citation-js).

> Note:
> 1. Browser tests don't work yet
> 1. Browser demo will be updated shortly
> 2. Loose years in BibTeX input don't work yet; use the `date` field

**Comment by larsgw** [comment]
I missed one reference to the old format... I'm really tied up in testing frameworks etc. (#55, #49) right now so fix might take some longer. Just don't sort by `accessed` or `issued` for now.

**Comment by larsgw** [comment]
Completely fixed now, loose years (and months) as well.

**Comment by rmzelle** [comment]
@larsgw, one of my users [reported](https://forums.zotero.org/discussion/comment/312445/#Comment_312445) an issue with BibTeX output missing date information (with citation.js v0.4.0-7). My input is CSL JSON. With CSL JSON as output format, the output is:

```json
[
  {
    "id": 454,
    "type": "article-journal",
    "title": "Abortion in the lives of women struggling financially: why insurance coverage matters",
    "container-title": "Guttmacher Policy Review",
    "page": "46–52",
    "volume": "19",
    "source": "Google Scholar",
    "shortTitle": "Abortion in the lives of women struggling financially",
    "author": [
      {
        "family": "Boonstra",
        "given": "Heather D."
      }
    ],
    "issued": {
      "date-parts": [
        [
          2016
        ]
      ]
    }
  }
]
```

With BibTeX as output format, the output is:

```
@article{Boonstra2016Abortion,
	journal={Guttmacher Policy Review},
	title={{Abortion in the lives of women struggling financially: why insurance coverage matters}},
	volume=19,
	author={Boonstra, Heather D.},
	pages={46–52},
}
```

As you can see, the "issued" date is missing from the BibTeX output. I do see dates in the BibTeX output for full dates (with year, month, and day), e.g.:

CSL JSON as output:

```json
  {
    "id": 459,
    "type": "report",
    "title": "State Funding of Abortion Under Medicaid",
    "publisher": "Guttmacher Institute",
    "URL": "https://www.guttmacher.org",
    "issued": {
      "date-parts": [
        [
          2018,
          2,
          1
        ]
      ]
    }
  }
]
```

BibTeX as output:

```
@techreport{2018State,
	publisher={Guttmacher Institute},
	title={{State Funding of Abortion Under Medicaid}},
	url={https://www.guttmacher.org},
	date={2018-02-01},
	year=2018,
	month=2,
	day=1,
}
```

**Comment by larsgw** [comment]
Previously, empty parts in the date were filled in with `1`, the default of the native date parser. This apparently breaks breaks some checks, and wasn't caught by test cases. I'm including a fix in the next release.

**Comment by larsgw** [comment]
Fixed in [`v0.4.0-8`](https://github.com/larsgw/citation.js/blob/master/CHANGELOG.md#040-8---2018-07-14). Now the output is:

```
@article{Boonstra2016Abortion,
	journal={Guttmacher Policy Review},
	title={{Abortion in the lives of women struggling financially: why insurance coverage matters}},
	volume=19,
	author={Boonstra, Heather D.},
	pages={46–52},
	date=2016,
	year=2016,
}
```

**Comment by rmzelle** [comment]
Thanks for the very quick fix!
